lib/core/src/sockComm.cpp

```diff
@@ -393,9 +388,44 @@ sockOpenForInConn( rsComm_t *rsComm, int *portNum, char **addr, int proto ) {
         return status;
     }
 
-    if ( addr != NULL ) {
-        *addr = ( char * )malloc( sizeof( char ) * LONG_NAME_LEN );
-        gethostname( *addr, LONG_NAME_LEN );
+    if (addr) {
+        namespace net = irods::experimental::net;
+
+        if (const auto entry = net::hnc_lookup("localhost"); entry) {
+            rodsLog(LOG_DEBUG, "%s :: Returning cached hostname for localhost [alias=%s].", __func__, entry->data());
+            *addr = static_cast<char*>(std::malloc(sizeof(char) * entry->size() + 1));
+            std::strncpy(*addr, entry->data(), entry->size());
+            (*addr)[entry->size()] = 0;
+        }
+        else {
+            rodsLog(LOG_NOTICE, "%s :: Hostname cache miss for localhost.", __func__);
+
+            constexpr auto size_in_bytes = sizeof(char) * LONG_NAME_LEN;
+            *addr = static_cast<char*>(std::malloc(size_in_bytes));
+            std::memset(*addr, 0, size_in_bytes);
+            gethostname(*addr, LONG_NAME_LEN);
+
+            try {
+                const auto& hosts_config = irods::get_server_property<nlohmann::json&>(irods::HOSTS_CONFIG_JSON_OBJECT_KW);
+                const auto alias = resolve_hostname(*addr, hosts_config, hostname_resolution_scheme::match_longest);
+
+                if (alias) {
+                    std::strncpy(*addr, alias->data(), alias->size());
+                }
+
+                const auto age = irods::get_server_property<std::chrono::seconds>(
+                    irods::configuration_parser::key_path_t{irods::CFG_HOSTNAME_CACHE_KW,
+                                                            irods::CFG_EVICTION_AGE_IN_SECONDS_KW});
+
+                const auto inserted = net::hnc_insert_or_assign("localhost", *addr, age);
```

Comment:
`*addr` contains the hostname obtained via `gethostname()`. The code presented will map `"localhost"` to the value of `gethostname()` if `resolve_hostname()` does not return an alias. This is so that the cache miss message is not logged on every invocation of this function.

Are we okay with this?

Reply:
This is running and being cached per server... so, yes, I think this is okay.

Reply:
cool